src/View/Helper/FormHelper.php

```diff
@@ -1176,11 +1178,15 @@ protected function _inputContainerTemplate(array $options): string
         if (!$this->templater()->get($inputContainerTemplate)) {
             $inputContainerTemplate = 'inputContainer' . $options['errorSuffix'];
         }
+        $requiredClassTemplate = $this->_defaultConfig['templates']['requiredClass'];
+        if ($this->templater()->get('requiredClass')) {
+            $requiredClassTemplate = $this->templater()->get('requiredClass');
+        }
```

Comment:
```suggestion
        $requiredClassTemplate = $this->templater()->get('requiredClass');
```

I don't think the condition adds anything, and if users want to remove the required class from their templates, they are free to do that. :shrug:

Reply:
It's for default config class `required`. For change just set template in AppView (tested and working):

```
$this->Form->setTemplates([
    'inputContainer' => /** @lang text */ '<div class="form__input --{{type}}{{required}}">{{content}}</div>',
    'requiredClass' => '--required',
]);
```

Reply:
Because class for backward comp. must be `required` as default. And if I want change default for my need then set custom class in my AppView.

Reply:
Sure, but the default template set also include a template for `requiredClass`?

Reply:
`$this->templater()->get('requiredClass');` without set template var in AppView return nothing :( I don't understant

Reply:
Aren't you including the `requiredClass in the `templates` configuration data? That is how all the other templates are defaulted and they are all read with `templater->get()`. This behaving differently means something is wrong with the current implementation.

Reply:
Any more inforamtion on this?

Reply:
It's done as @markstory sugested...